### DISCUSSION TITLE
Clip Arc To Cell

### DISCUSSION THREAD

**Discussion by r-barnes** [google/s2geometry#346] [Q&A]
Is there a way to clip a great circle arc to a cell?

I see two potential approaches.

1. Use `S2::GetIntersection` to find intersections between cell edges and the arc. Perform additional checks to make sure that the intersection is actually on the arcs (since it appears GetIntersection returns a nearby great circle intersection). Choose two intersections appropriately.

2. Use `bool ClipEdge(const R2Point& a, const R2Point& b, const R2Rect& clip, R2Point* a_clipped, R2Point* b_clipped);` somehow; however, this requires projecting the cell and the arc into the uv plane and I'm not sure what the precision loss will be.

**Comment by jmr** [comment]
Again, what is the higher level problem you're trying to solve? It seems like you might be working at too low a level. There might be other tools that solve your problem, but I don't know what your real problem is.
